__tests__/commands/install/integration.js

```diff
@@ -37,6 +56,21 @@ test.concurrent('properly find and save build artifacts', async () => {
   });
 });
 
+test('changes the cache path when bumping the cache version', async () => {
+  await runInstall({}, 'install-github', async (config): Promise<void> => {
+    const inOut = new stream.PassThrough();
+    const reporter = new reporters.JSONReporter({stdout: inOut});
+
+    await cache(config, reporter, {}, ['dir']);
+    assert.ok(!!(JSON.parse(String(inOut.read())) : any).data.match(/\/v1\/?$/));
+
+    await mockConstants({CACHE_VERSION: 42}, async (config): Promise<void> => {
+      await cache(config, reporter, {}, ['dir']);
+      assert.ok(!!(JSON.parse(String(inOut.read())) : any).data.match(/\/v42\/?$/));
```

Comment:
I think you should add a more high level check that verifies connection between current CACHE_VERSION and what gets installed in node_modules.

E.g.
1. Install package-a
2. add am empty file MARKER to the cache folder of package-a
3. install package-a (with --skip-integrity-check flag) and verify that MARKER was copied
4. set CACHE_VERSION=42
5. install package-a (with --skip-integrity-check flag) and verify that MARKER was removed in node_modules
6. set CACHE_VERSION=1
7. install package-a (with --skip-integrity-check flag) and verify that MARKER was copied